### PR TITLE
Add shared_span_context to zipkin tracer API

### PR DESCRIPTION
**Pull request by bianpengyuan**
Context: https://github.com/istio/istio/issues/13508#issuecomment-485689990

This is to make shared_span_context in envoy's zipkin tracer configurable (https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/trace/v2/trace.proto#config-trace-v2-zipkinconfig)

cc @objectiser @douglas-reid ptal thanks!

### CODE REVIEW

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -43,6 +43,12 @@ message Tracing {
   message Zipkin {
     // Address of the Zipkin service (e.g. _zipkin:9411_).
     string address = 1;
+
+    // Control weather to turn on shared_span mode for Zipkin span generation.
```
Comment (objectiser) [inline mesh/v1alpha1/proxy.proto]: nit: "whether" instead of "weather"

**Comment by douglas-reid** [comment]
I'd favor using the workaround of custom bootstrap CM + annotation until we have proper tracing support via API.

I'm concerned that this goes against the move away from putting more into mesh config.

File: mesh/v1alpha1/proxy.proto [hunk 2]
```diff
@@ -43,6 +43,12 @@ message Tracing {
   message Zipkin {
     // Address of the Zipkin service (e.g. _zipkin:9411_).
     string address = 1;
+
+    // Controls whether to turn on shared_span mode for Zipkin span generation.
+    // If this is true, shared_span mode will be enabled and envoy will generate
+    // a new child span for ingress request. If this is false, share_span mode will
+    // be turned off, such that client and server span will share the same span id.
+    bool shared_span_context = 2;
```
Comment (costinm) [inline mesh/v1alpha1/proxy.proto]: @ostromart @douglas-reid 
/hold 
This conflicts with the plans for operator and I believe there are proposals to refactor/cleanup the tracing settings.
Most of them are in values.yaml, some are telemetry, etc.

Let's not add to the mess.

**Comment by bianpengyuan** [comment]
Close as now the plan is to avoid adding more knob into mesh config. Instead tracing api or operator should support this in the future.

**Pull request closed without merging**